### DISCUSSION TITLE
Feature Request: ShouldBeEquivalentTo(IEnumerable) and ShouldNotBeEquivalentTo(IEnumerable)

### DISCUSSION THREAD

**Discussion by CreepyGnome** [shouldly/shouldly#784] [Ideas]
I can't seem to find a Shouldly "Should" that would assert if two IEnumerable objects are equivalent to each other by determining they contain the same items regardless of order.

Maybe have another Should that requires them to be in the same order, maybe add an override to ShouldBeSameAs and ShouldNotBeSameAs for IEnumerable objects.

**Comment by CreepyGnome** [comment]
Missed it, the ShouldBe method has an override seems to account for this scenario, however, just for feedback this isn't really clear and when looking through the documentation didn't seem to find it. I found it by looking through your source. I would never have thought ShouldBe would do this since it normal compares Equality not Equivalency form what I could tell. But still new to Shouldly so what do I know. 

The name doesn't feel right, because comparing two different list references to see if they have the same contents doesn't sound like "ShouldBe" grammar to me and not clear when I am using it. To me it seems my suggested naming is more clear to the purpose "ShouldBeEquivalentTo" is more precises. To me "ShouldBe" sounds like it is just going to make sure they are the same reference.

Again what do I know, I am brand new to using Shouldly. Closing the issue.

**Comment by JakeGinnivan** [comment]
Hey, for reference equality we have `ShouldBeSameAs`. We are working on moving the docs over to readthedocs now which hopefully should make this a bit more clear. Pull requests welcome.

The thing is if I go `list.ShouldBe(new[] { '', '', ''});` the enumerable types are clearly different so i would never want reference equality. Happy to continue this conversation though, anything to make shouldly even more intuitive is good!

**Comment by CreepyGnome** [comment]
I would just say you should never assume what others will want to do with Shouldly. Providing for content equivalency and reference equivalency are just features or tools in the Shouldly toolbox. How people use it is up to them and they may have good reasons to do something you may not think is correctly.

Also I am merely pointing out the semantics of the words uses, as "ShouldBe" is very strong and to me seems like it should be the same as "object == object" or "primitive == primitive" and "ShouldBeSameAs" seems a little bit on the redundant side. 

One thing that drew me to look at Shouldly was the apparent of nice language constructs for asserts. However I started to get confused with some of them as not being as clear as I thought Shouldly should be. (Pun intended)  Seems like "ShouldBeEquivalentTo" or even something like "ShouldHaveEquivalentContenTo" seem to make it more clear to me. Again, these are just my thoughts and I could be way off base.  

I am still using it now in new projects as it is still much better that the built in Assert.XXX in testing frameworks. I am just getting over the learning curve I think with the Shouldy libraries opinions and that is all.  Thanks for the response.

**Comment by xhafan** [comment]
Could you please advise what is the Shoudly equivalent for NUnit's  `Is.EquivalentTo`:
```c#
Assert.That(new[] { "string one", "string two" }, Is.EquivalentTo(new[] { "string two", "string one" }));
```
The above passes, but the below does not pass:
```c#
new[] { "string one", "string two" }.ShouldBe(new[] { "string two", "string one" });
```
as Shouldly matches string one by one and the order is important.

Collection comparison regardless the order is actually the only case when I still use NUnit's `Assert.That`

Edit: `ShouldBeEquivalentTo` is now implemented in [4.0.0-beta0002](https://github.com/shouldly/shouldly/releases/tag/4.0.0-beta0002) release https://github.com/shouldly/shouldly/pull/411#issuecomment-502581222 , but is not quite as useful as `Assert.That(...,  Is.EquivalentTo` when comparing collections.

**Comment by xhafan** [comment]
@CreepyGnome Why did you close this issue? As far as I know Shouldly is not able to compare collections regardless the order of items. Could you please re-open it?

**Comment by jnm2** [comment]
@xhafan This is how Shouldly compares collections regardless of order:

```cs
new[] { 1, 2 }.ShouldBe(new[] { 2, 1 }, ignoreOrder: true);
```

**Comment by josephwoodward** [comment]
@xhafan @jnm2 is correct, the example [can be found here](https://github.com/shouldly/shouldly/blob/master/src/Shouldly.Tests/ShouldBe/EnumerableType/IgnoreOrderScenario.cs#L52).

I'm currently working on improving the docs so watch this space.
